src/Microsoft.Management.Configuration/ConfigurationSetSerializer.cpp

```diff
@@ -46,21 +46,35 @@ namespace winrt::Microsoft::Management::Configuration::implementation
 
         for (const auto& [key, value] : valueSet)
         {
-            IPropertyValue property = value.try_as<IPropertyValue>();
             std::string keyName = winrt::to_string(key);
-            auto type = property.Type();
+            const auto& currentValueSet = value.try_as<Windows::Foundation::Collections::ValueSet>();
 
-            if (type == PropertyType::Boolean)
+            if (currentValueSet)
             {
-                emitter << AppInstaller::YAML::Key << keyName << AppInstaller::YAML::Value << property.GetBoolean();
+                emitter << AppInstaller::YAML::Key << keyName;
+                WriteYamlValueSet(emitter, currentValueSet);
             }
-            else if (type == PropertyType::String)
+            else
             {
-                emitter << AppInstaller::YAML::Key << keyName << AppInstaller::YAML::Value << AppInstaller::Utility::ConvertToUTF8(property.GetString());
-            }
-            else if (type == PropertyType::Int64)
-            {
-                emitter << AppInstaller::YAML::Key << keyName << AppInstaller::YAML::Value << property.GetInt64();
+                IPropertyValue property = value.try_as<IPropertyValue>();
+                auto type = property.Type();

```

Comment:
I don't know how strict `ValueSet` is at preventing anything that isn't a `ValueSet` or `IPropertyValue` from getting in, so this could result in a null dereference.  Either change the `try_as` to just `as` or check for null and throw.

Reply:
Changed IPropertyValue conversion to using `as`